### PR TITLE
docs: update Readme and add script to generate API docs

### PR DESCRIPTION
**Pull request by arjungarg07**
This PR updates Readme and add more information regarding:

- example usage of the library
- every single diagram syntax and example output screen shot
- steps that user needs to perform to get a final diagram rendered for react flow
- explanation of technical details of how the relations are discovered.
- what is in the default output map, what info is available

Also, added script to generate API docs instead of manual maintainance
**Related issue(s)**
#29

### CODE REVIEW

**Comment by arjungarg07** [comment]
@derberg @magicmatatjahu need a review here 😅

File: README.md [hunk 1]
```diff
@@ -38,4 +67,196 @@ const plantUMLClassDiagram = appRelationsDiscovery.getRelations(...docs,{syntax:
...
+server1 -- flight/queue --> FlightMonitorService
+```
+
+### Flowchart
+
+<img src="./public/img/mermaid.png" alt="Mermaid Flowchart" width="500"/> </br></br>
+
+### Link
+
+You can paste the mermaid output syntax to visualize the relations as Flowchart here: https://mermaid-js.github.io/mermaid-live-editor/edit
```
Comment (derberg) [inline README.md:139]: I think editor supports you to put all data encoded in the url so you can already provide a direct link here. Also, don't call it `Link` section but just mention below code snippet with syntac

File: README.md [hunk 2]
```diff
@@ -38,4 +67,196 @@ const plantUMLClassDiagram = appRelationsDiscovery.getRelations(...docs,{syntax:
...
+FlightSubscriberService --|> server1:flight/queue
+server1 --|> FlightMonitorService:flight/queue
+@enduml
+```
+### ClassDiagram
+
+<img src="./public/img/plantUML.png" alt="PlantUML classDiagram" width="500"/> </br></br>
+### Link
+
+You can paste the plantUML output syntax to visualize the relations as classDiagram here: https://www.planttext.com/
```
Comment (derberg) [inline README.md:163]: please mention below code snippen, no Link section. Also check if there is the same possibility as with Mermaid (encoding)

File: README.md [hunk 3]
```diff
@@ -38,4 +67,196 @@ const plantUMLClassDiagram = appRelationsDiscovery.getRelations(...docs,{syntax:
...
+
+### Default Output Syntax
+
+// Explain:
+
+```javascript
+Map(n) {
+  '<server1>' => Map(m) {
+    'channel1' => { sub: [Map], pub: [Map] },
+    'channel2' => { sub: [Map], pub: [Map] }
```
Comment (derberg) [inline README.md:114]: how the map looks like? please expand

**Review by derberg — COMMENTED** [review]
some comments 😅

File: README.md [hunk 4]
```diff
@@ -9,16 +9,45 @@
...
 ```
 
-## API
+## Testing
+1) Clone the project `git clone https://github.com/asyncapi/app-relations-discovery.git`
+2) Install the dependencies `npm i`
+3) For a quick overview you can run tests by `npm test`. You can also contribute to provide more different syntax outputs to visualize the relations.
+
+### Technical Details
+This library takes AsyncAPI Documents as an array input for which the user wants to discover the relations of. It then validates and parse given array of AsyncAPI documents and generates the output in desired passed syntax.
```
[the remark this answers is not in the document]
Comment (arjungarg07) [reply README.md:48]: I liked the idea of not adding the responsibility to this package, will follow up on this one.

File: README.md [hunk 5]
```diff
@@ -7,35 +7,230 @@
 
 ![npm](https://img.shields.io/npm/v/@asyncapi/app-relations-discovery?style=for-the-badge) ![npm](https://img.shields.io/npm/dt/@asyncapi/app-relations-discovery?style=for-the-badge)
 
-> :warning: This package is under construction.
-
+- [Overview](#overview)
```
Comment (derberg) [inline README.md:10]:
```suggestion
<!-- toc is generated with GitHub Actions do not remove toc markers -->

<!-- toc -->

- [Overview](#overview)
```

File: README.md [hunk 6]
```diff
@@ -7,35 +7,230 @@
...
+  - [Flowchart](#flowchart)
+- [PlantUML classDiagram](#plantuml-classdiagram)
+  - [Syntax](#syntax-1)
+  - [ClassDiagram](#classdiagram)
+- [React Flow  Nodes](#react-flow--nodes)
+  - [Syntax](#syntax-2)
+  - [React Flow Nodes](#react-flow-nodes)
+  - [Steps to visualize relations in React Flow](#steps-to-visualize-relations-in-react-flow)
+- [Develop](#develop)
+- [Contributing](#contributing)
```
Comment (derberg) [inline README.md:28]:
```suggestion
- [Contributing](#contributing)

<!-- tocstop -->

```

File: README.md [hunk 7]
```diff
@@ -7,35 +7,230 @@
...
+FlightSubscriberService[Flight Subscriber Service]
+FlightSubscriberService -- flight/queue --> server1
+server1 -- flight/queue --> FlightMonitorService
+```
+
+### Flowchart
+
+<img src="./public/img/mermaid.png" alt="Mermaid Flowchart" width="500"/> </br></br>
+
+**[Code snippet with syntax](https://mermaid-js.github.io/mermaid-live-editor/edit#eyJjb2RlIjoiZ3JhcGggVERcbiBzZXJ2ZXIxWyhtcXR0Oi8vbG9jYWxob3N0OjE4ODMpXVxuRmxpZ2h0TW9uaXRvclNlcnZpY2VbRmxpZ2h0IE1vbml0b3IgU2VydmljZV1cbkZsaWdodE1vbml0b3JTZXJ2aWNlIC0tIGZsaWdodC91cGRhdGUgLS0-IHNlcnZlcjFcbkZsaWdodE5vdGlmaWVyU2VydmljZVtGbGlnaHQgTm90aWZpZXIgU2VydmljZV1cbnNlcnZlcjEgLS0gZmxpZ2h0L3VwZGF0ZSAtLT4gRmxpZ2h0Tm90aWZpZXJTZXJ2aWNlXG5GbGlnaHRTdWJzY3JpYmVyU2VydmljZVtGbGlnaHQgU3Vic2NyaWJlciBTZXJ2aWNlXVxuRmxpZ2h0U3Vic2NyaWJlclNlcnZpY2UgLS0gZmxpZ2h0L3F1ZXVlIC0tPiBzZXJ2ZXIxXG5zZXJ2ZXIxIC0tIGZsaWdodC9xdWV1ZSAtLT4gRmxpZ2h0TW9uaXRvclNlcnZpY2UiLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjp0cnVlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6dHJ1ZX0)**
```
Comment (derberg) [inline README.md:111]:
```suggestion
**[View in online editor](https://mermaid-js.github.io/mermaid-live-editor/edit#eyJjb2RlIjoiZ3JhcGggVERcbiBzZXJ2ZXIxWyhtcXR0Oi8vbG9jYWxob3N0OjE4ODMpXVxuRmxpZ2h0TW9uaXRvclNlcnZpY2VbRmxpZ2h0IE1vbml0b3IgU2VydmljZV1cbkZsaWdodE1vbml0b3JTZXJ2aWNlIC0tIGZsaWdodC91cGRhdGUgLS0-IHNlcnZlcjFcbkZsaWdodE5vdGlmaWVyU2VydmljZVtGbGlnaHQgTm90aWZpZXIgU2VydmljZV1cbnNlcnZlcjEgLS0gZmxpZ2h0L3VwZGF0ZSAtLT4gRmxpZ2h0Tm90aWZpZXJTZXJ2aWNlXG5GbGlnaHRTdWJzY3JpYmVyU2VydmljZVtGbGlnaHQgU3Vic2NyaWJlciBTZXJ2aWNlXVxuRmxpZ2h0U3Vic2NyaWJlclNlcnZpY2UgLS0gZmxpZ2h0L3F1ZXVlIC0tPiBzZXJ2ZXIxXG5zZXJ2ZXIxIC0tIGZsaWdodC9xdWV1ZSAtLT4gRmxpZ2h0TW9uaXRvclNlcnZpY2UiLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjp0cnVlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6dHJ1ZX0)**
```

File: README.md [hunk 8]
```diff
@@ -7,35 +7,230 @@
...
+FlightSubscriberService --|> server1:flight/queue
+server1 --|> FlightMonitorService:flight/queue
+@enduml
 ```
+### ClassDiagram
 
-- For plantUML classDiagram 
+<img src="./public/img/plantUML.png" alt="PlantUML classDiagram" width="500"/> </br></br>
+
+**[Code snippet with syntax](https://www.planttext.com/?text=VP3D3e8m3CVlVOgz02742zd1a6YywOKd6AF0aa6Xwxoexou1EM2CjpRz_JzIA88ObjXx42SUrScR432eP9tKsPcMJGzWbpKWtv4pzL2W8dkj-ab4fwadQtn7GNIMvuVvE389MVeGy8ABTsqdLngS49UpZREeakHvt3nrin1f76iZ25lIWgpY6ubh76xgBy7AbB4Abbs5VpEcYF5dnBxV7YzmgET7lG40)**
```
Comment (derberg) [inline README.md:134]:
```suggestion
**[View in online editor](https://www.planttext.com/?text=VP3D3e8m3CVlVOgz02742zd1a6YywOKd6AF0aa6Xwxoexou1EM2CjpRz_JzIA88ObjXx42SUrScR432eP9tKsPcMJGzWbpKWtv4pzL2W8dkj-ab4fwadQtn7GNIMvuVvE389MVeGy8ABTsqdLngS49UpZREeakHvt3nrin1f76iZ25lIWgpY6ubh76xgBy7AbB4Abbs5VpEcYF5dnBxV7YzmgET7lG40)**
```

File: API.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+## getRelations(asyncApiDocs, [options]) ⇒ <code>Promise.&lt;DiscoveredRelations&gt;</code>
+Validates and analyzes a list of AsyncAPI documents and get applications described by those files
+
+**Kind**: global function  
+**Returns**: <code>Promise.&lt;DiscoveredRelations&gt;</code> - Relations between documents  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| asyncApiDocs | <code>Array</code> | An array of asyncApiDocuments |
```
Comment (derberg) [inline API.md:31]: specify in jsdoc that it is an array of strings

File: package.json [hunk 10]
```diff
@@ -6,9 +6,11 @@
   "scripts": {
     "test": "npm run test:lib",
     "lint": "eslint --max-warnings 0 --config .eslintrc .",
+    "docs": "jsdoc2md \"lib/appRelationsDiscovery.js\" -f \"lib/**/*.js\" > API.md",
     "get:version": "echo $npm_package_version",
     "get:name": "echo $npm_package_name",
-    "generate:assets": "echo \"Nothing to generate\"",
+    "generate:readme:toc": "markdown-toc -i README.md",
+    "generate:assets": "npm run docs",
```
Comment (derberg) [inline package.json:13]: toc generation should be done here too

**Review by derberg — COMMENTED** [review]
some tiny fixes 😉

File: README.md [hunk 11]
```diff
@@ -7,35 +7,237 @@
...
 ## Install
 
 ```
 npm install @asyncapi/app-relations-discovery
 ```
 
-## API
+### Technical Details
+This library takes AsyncAPI files as an array input for which the user wants to discover the relations between them. It then validates and parses the given array of AsyncAPI files and generates the output in desired passed syntax.
+In the process, for every different server it assigns a slug having server's URL and protocol and then maps channels with the same server. Following, it maps the service information with the channel's name as per if the service is subscribing/publishing to a given channel. In the sub/pub Map of default output syntax, it provides service name and the metadata of the service including but not limited to `description`, `payload`, `headers`, `bindings`, `extensions`.
```
Comment (derberg) [inline README.md:46]:
```suggestion
In the process, for every different server, it assigns a slug having the server's URL and protocol and then maps channels with the same server. Following, it maps the service information with the channel's name as per if the service is subscribing/publishing to a given channel. In the sub/pub Map of default output syntax, it provides service name and the metadata of the service including but not limited to `description`, `payload`, `headers`, `bindings`, `extensions`.
```

File: lib/appRelationsDiscovery.js [hunk 12]
```diff
@@ -11,7 +11,7 @@ const defaultOptions = {
 /**
  * Validates and analyzes a list of AsyncAPI documents and get applications described by those files
  * 
- * @param {Array} asyncApiDocs An array of asyncApiDocuments
+ * @param {Array} asyncApiDocs An array of strings
```
Comment (derberg) [inline lib/appRelationsDiscovery.js:14]:
```suggestion
 * @param {String[]} asyncApiDocs An array of stringified AsyncAPI documents
```

this is what I meant, I'm not 100% sure on the jsdoc syntax, you will have to check how it generates into API.md

**Review by derberg — COMMENTED** [review]
@arjungarg07 super minor last 2 comments

File: lib/appRelationsDiscovery.js (hunk 12 above)
Comment (arjungarg07) [reply lib/appRelationsDiscovery.js]: Updated manually.

**Comment by derberg** [comment]
@arjungarg07 was coverage workflow failing  before? I don't recall

**Comment by arjungarg07** [comment]
@derberg Yeah, coverage workflow failing since added 😅

**Comment by arjungarg07** [comment]
Nope, all good

**Pull request merged**